jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -525,7 +525,11 @@ private void checkAccess(Class<?> definingClass, Class<?> referenceClass, String
 			}
 			/*[MSG "K0587", "'{0}' no access to: '{1}'"]*/
 			String errorMessage = com.ibm.oti.util.Msg.getString("K0587", this.toString(), definingClass.getName() + extraInfo);  //$NON-NLS-1$
-			throw new IllegalAccessException(errorMessage);
+
+			/* Error should be IllegalAccessException caused by IllegalAccessError */
+			IllegalAccessException iae = new IllegalAccessException();
+			iae.initCause(new IllegalAccessError(errorMessage));
+			throw iae;
```

Comment:
This feels really hacky and we shouldn't do this.  Any test requiring this is bogus.